js/modules/require_impl.go

```diff
@@ -67,3 +74,81 @@ func (r *LegacyRequireImpl) Require(specifier string) (*goja.Object, error) {
 func (r *LegacyRequireImpl) CurrentlyRequiredModule() url.URL {
 	return *r.currentlyRequiredModule
 }
+
+func (r *LegacyRequireImpl) warnUserOnPathResolutionDifferences(specifier string) {
+	normalizePathToURL := func(path string) (*url.URL, error) {
+		u, err := url.Parse(path)
+		if err != nil {
+			return nil, err
+		}
+		return loader.Dir(u), nil
+	}
+	// Warn users on their require depending on the none standard k6 behaviour.
+	rt := r.vu.Runtime()
+	logger := r.vu.InitEnv().Logger
+	correct, err := normalizePathToURL(getCurrentModuleScript(rt))
+	if err != nil {
+		logger.Warningf("Couldn't get the \"correct\" path to resolve specifier %q against: %q"+
+			"Please report to issue %s. "+
+			"This will not currently break your script but might mean the future it won't work",
+			specifier, err, issueLink)
+	} else if r.currentlyRequiredModule.String() != correct.String() {
+		logger.Warningf("The \"wrong\" path (%q) and the actually used by k6 path (%q) to resolve %q are different. "+
+			"Please report to issue %s. "+
+			"This will not currently break your script but *WILL* in the future, please report this!!!",
+			correct, r.currentlyRequiredModule, specifier, issueLink)
+	}
+
+	k6behaviourString, err := getPreviousRequiringFile(rt)
+	if err != nil {
+		logger.Warningf("Couldn't get the \"wrong\" path to resolve specifier %q against: %q"+
+			"Please report to issue %s. "+
+			"This will not currently break your script but might mean the future it won't work",
+			specifier, err, issueLink)
+		return
+	}
+	k6behaviour, err := normalizePathToURL(k6behaviourString)
+	if err != nil {
+		logger.Warningf("Couldn't get the \"wrong\" path to resolve specifier %q against: %q"+
+			"Please report to issue %s. "+
+			"This will not currently break your script but might mean the future it won't work",
+			specifier, err, issueLink)
+		return
+	}
+	if r.currentlyRequiredModule.String() != k6behaviour.String() {
+		// this should always be equal, but check anyway to be certain we won't break something
+		logger.Warningf("The \"wrong\" path (%q) and the actually used by k6 path (%q) to resolve %q are different. "+
+			"Please report to issue %s. "+
+			"This will not currently break your script but might mean the future it won't work",
+			k6behaviour, r.currentlyRequiredModule, specifier, issueLink)
+	}
+}
+
+func getCurrentModuleScript(rt *goja.Runtime) string {
+	var parent string
+	var buf [2]goja.StackFrame
+	frames := rt.CaptureCallStack(2, buf[:0])
+	parent = frames[1].SrcName()
```

Comment:
Is this always safe? Is there any change to get `frames` with only one or no items at all?🤔

Reply:
In all of the experiments I have ran in the last 4 months or so (this particular code is very old) I have had 1 panic when I called it when there was nothing running on the runtime. 

As I would argue that is very much the wrong way to use it ... I am particularly certain it makes sense to capture this it seems.